pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -806,6 +809,54 @@ func TestDesiredRouterDeploymentSpecAndNetwork(t *testing.T) {
 	}
 
 	checkDeploymentHasEnvSorted(t, deployment)
+
+	// Set max dynamic server when DCM feature gate is on.
+	ic.Spec.UnsupportedConfigOverrides = runtime.RawExtension{
+		Raw: []byte(`{"maxDynamicServers":"7"}`),
+	}
+	deployment, err = desiredRouterDeployment(ic, ingressControllerImage, ingressConfig, infraConfig, apiConfig, networkConfig, proxyNeeded, false, nil, clusterProxyConfig, false, true)
+	if err != nil {
+		t.Fatalf("invalid router Deployment: %v", err)
+	}
+	tests = []envData{
+		{"ROUTER_HAPROXY_CONFIG_MANAGER", true, "true"},
+		{"ROUTER_MAX_DYNAMIC_SERVERS", true, "7"},
+	}
+	if err := checkDeploymentEnvironment(t, deployment, tests); err != nil {
+		t.Error(err)
+	}
+
+	// Set max dynamic server to a wrong value when DCM feature gate is on.
+	ic.Spec.UnsupportedConfigOverrides = runtime.RawExtension{
+		Raw: []byte(`{"maxDynamicServers":"wrong"}`),
+	}
+	deployment, err = desiredRouterDeployment(ic, ingressControllerImage, ingressConfig, infraConfig, apiConfig, networkConfig, proxyNeeded, false, nil, clusterProxyConfig, false, true)
+	if err != nil {
+		t.Fatalf("invalid router Deployment: %v", err)
+	}
+	tests = []envData{
+		{"ROUTER_HAPROXY_CONFIG_MANAGER", true, "true"},
+		{"ROUTER_MAX_DYNAMIC_SERVERS", true, "1"},
+	}
+	if err := checkDeploymentEnvironment(t, deployment, tests); err != nil {
+		t.Error(err)
+	}
+
+	// Opt out from DCM via unsupportedConfigOverride.
+	ic.Spec.UnsupportedConfigOverrides = runtime.RawExtension{
+		Raw: []byte(`{"dynamicConfigManager":"false", "maxDynamicServers":"7"}`),
+	}
+	deployment, err = desiredRouterDeployment(ic, ingressControllerImage, ingressConfig, infraConfig, apiConfig, networkConfig, proxyNeeded, false, nil, clusterProxyConfig, false, true)
+	if err != nil {
+		t.Fatalf("invalid router Deployment: %v", err)
+	}
+	tests = []envData{
+		{"ROUTER_HAPROXY_CONFIG_MANAGER", false, ""},
+		{"ROUTER_MAX_DYNAMIC_SERVERS", false, "0"},
+	}
+	if err := checkDeploymentEnvironment(t, deployment, tests); err != nil {
+		t.Error(err)
+	}
```

Comment:
Long ago, we had just one gigantic `TestDesiredRouterDeployment` unit test.  This test became difficult to modify when we needed to expand test coverage, and so in https://github.com/openshift/cluster-ingress-operator/pull/712/commits/2849fcf835669b5c1f9d8ab21d6195287b0a5502, Candace started splitting that gigantic test into multiple tests.  We should continue in that direction: more, smaller, focused tests rather than fewer, larger, "all the things!" tests.  

What do you think about defining a new `TestDesiredRouterDeploymentDynamicConfigManager` test, perhaps using a list of test cases à la `TestDesiredRouterDeploymentDefaultPlacement`?

Reply:
> What do you think about defining a new TestDesiredRouterDeploymentDynamicConfigManager test, perhaps using a list of test cases à la TestDesiredRouterDeploymentDefaultPlacement?

> Upd: This test already exists: TestUnsupportedConfigOverride.

At the end, I think the best way would be to make it a dedicated unit test indeed. Working on it.

Reply:
`TestDesiredRouterDeploymentDynamicConfigManager` unit test added.